### ISSUE TITLE
[nms_Arpley] Player stuck spawning in rails

### ISSUE THREAD

**Issue by Wazanator** [ZPSNMRiHIssues/nmrihissues#733]
http://steamcommunity.com/app/224260/discussions/0/617336568069805109/

http://i.imgur.com/ux9YlrG.png

Players can sometimes accidentally get stuck in rails upon spawning.

**Comment by Fersigal** [comment]
@Bubka3  See https://forums.nomoreroominhell.com/index.php?/topic/12183-patch-1094-changelog/
Should be fixed, will have to ask @nay0r

**Comment by AtheistNMRiH** [comment]
1.09.4 posted april 17 -> this issue is posted 22 may, so I'm not sure that it is fixed
and it says "bridge support beams", and here are rails

**Comment by AtheistNMRiH** [comment]
actually rails are not solid (even in older versions), player stuck in Clip brush, @Bubka3 or @felis-catus do you know if casual spawn system check player spawn spot for the presence of clips? looks like game spawns players fine in not solid prop, but it doesn't know, that there is a clip

**Comment by Fersigal** [comment]
I do actually remember a spawning bug on casual on nmo_shelter as of recently, whereas a player was spectating another player and it spawned to the ground but there was a clip underneath it and the player would be stuck there. I think it happened at the connector hall with a fence and there was a clip underneath it, but because someone was spectating he spawned in the person but it registered as if it was underneath the player, so it spawned it inside the red collision clip (for players and zombies that is).

**Comment by Bubka3** [comment]
I think the casual system isnt checking for clips. Felis will check this.

**Comment by Bubka3** [comment]
@AtheistNMRiH are any of the spawn points in a clip or brush?

**Comment by AtheistNMRiH** [comment]
no, I guess it happens when player spawns in radius of another player

**Comment by Fersigal** [comment]
It happens when a "dead" player which is about to respawn spectates another player and there's enough space for the player to spawn (no zombies nearby) and it spawns on top of where the player is currently running (spawntoground) like a normal player spawn point. If this would happen onto a normal player spawn this problem will also occur and it will also spawn inside the brush. It's simply a code bug, an easy fix would be to take clip brushes into consideration so it doesn't spawn inside the brush.

**Comment by felis-catus** [comment]
Added additional check for player clips in code commit r4311, not sure if it resolves this. Needs more testing.

**Comment by Winning117** [comment]
Title typo "causal" instead of "casual", should be fixed for easier tracking.

**Comment by felis-catus** [comment]
Still occurring in 1.11.4.
